### PR TITLE
Development Guidelines for the Lab

### PR DESCRIPTION
**Pull request by awwad**
Create a central location for development guidelines for the lab. A variety of documents exists for the Seattle project that should be drawn here, and some of our unwritten practices need to be codified as well.

### CODE REVIEW

File: development_workflow.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Development / Contribution Guidelines
+
+## General Development Workflow in the Lab
+
+
+
+### Forking and Branching
```
Comment (vladimir-v-diaz) [inline development_workflow.md]: We should link to our [introduction to GitHub/git guide](https://github.com/SeattleTestbed/docs/blob/master/Archive/Local/RepoAccess.md) for newcomer to the lab.  According to Albert's [comment in another issue](https://github.com/secure-systems-lab/code-style-guidelines/issues/6#issuecomment-285608378), the images are missing but have been gitified.

Comment (aaaaalbert) [inline development_workflow.md]: I'd go the other way around and rather move that guide from the Seattle docs repo into here.

Comment (aaaaalbert) [inline development_workflow.md]: Pics: https://github.com/SeattleTestbed/docs/tree/master/ATTACHMENTS/Local/RepoAccess

File: development_workflow.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+The lab's general workflow for contributing to a project involves:
```
Comment (vladimir-v-diaz) [inline development_workflow.md]: We should mention that PRs must not be merged into the "master" branch with build failures or lack unit tests.  

Another subject worth discussing is what happens if a developer is working on a large feature that must be broken into multiple parts/PRs  (to minimize the size of the PR that others must review).

In this case, if the developer wishes to work incrementally on a large feature, the project maintainer(s) might (1) create a branch on the main repository for that feature (2) have the developer fork from that branch (3) contributors can then issue multiple, smaller PRs to that feature branch.  The smaller PRs to the feature branch do not necessarily need to build successfully or be unit tested.  The goal of this workflow is to encourage smaller PRs, consistent reviews of PRs, and when the feature branch is fully implemented and tested it is simply merged with master.

File: development_workflow.md (hunk 1 above)
Comment (awwad) [inline development_workflow.md]: Agreed: migrate here.

File: development_workflow.md (hunk 2 above)
Comment (awwad) [inline development_workflow.md]: Yes, we have to add the solutions to splitting PRs up that we've discussed. We should add that to the section below on PRs. This section is meant to be a quick overview.

**Comment by awwad** [comment]
I think that makes sense. And no, of course I don't. ^_^

**Comment by lukpueh** [comment]
This PR was moved to a different repo. Please review/merge/move forward in https://github.com/secure-systems-lab/lab-guidelines/pull/4.

**Pull request closed without merging**